src/main/python/analyzers/project_analyzer.py

```diff
@@ -62,16 +87,60 @@ def __performRegressionAnalysis(self, dfWithFeatures, featureColumnIndexes):
         @rtype: dictionary
         """
         # Perform regression analysis
-        # return resultsDictOrDF
+        X = dfWithFeatures[featureColumnIndexes]
+        y = dfWithFeatures['PlusMinusPoints']
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
+
+        # Generating models
+        lr = LinearRegression()
+        lr.fit(X_train, y_train)
+        prediction = lr.predict(X_test)
+        resultsDictOrDF = pd.concat([y_test, pd.Series(prediction, index=y_test.index, name='y_test')], axis=1)
+        resultsDictOrDF.columns = ['y_pred', 'y_test']
```

Comment:
Oh wait, it this saying to only return a dataframe with tweetID as the index and then these columns? If so, nice!

Reply:
yes, that's the case. the concat doesn't work if the elements don't share the same index (similar to changes in src/main/python/analyzers/emotion_analyzer.py - see my comment there)